bukuserver/templates/bukuserver/bookmarks_list.html

```diff
@@ -6,9 +6,22 @@
   {{ buku.close_if_popup() }}
 {% endblock %}
 
+{% block model_menu_bar_before_filters %}
+  {{ super() }}
+  {% if data %}
+    {% set _random = url_for('bookmark.details_view', modal=True, id='random', url=return_url, **(request.args|flt)) %}
+    <li><a id="random" data-target="#fa_modal_window" data-toggle="modal" href="{{ _random }}">Random</a></li>
+  {% endif %}
+{% endblock %}
+
 {% block tail %}
   {{ super() }}
   {{ buku.script('buku_filter.js') }}
   {{ buku.focus(None) }}
   {{ buku.link_saved() }}
+  <script>
+    $(document).on('click', `#modal-random`, function() {
+      $(`#fa_modal_window .modal-content`).load($(`#random`).attr('href'));
+    });
+  </script>
```

Comment:
Implementing “Pick another” here (this code is only run once so there's no danger of a livelock/memory leak).